### PR TITLE
Issue #17: PR evidence standard — fixture corpus, golden tests, preview script

### LINKED ISSUE #17: [P1] Add agentic PR evidence standard: committed fixture corpus, golden outputs, and preview artifact
This repo is now running heavily agentic implementation across ingestion and follow-up pipeline work. Code arrives quickly, but PRs are currently hard to review for *behavior*:

- \`data/raw/\`, \`data/extracted/\`, \`data/canonical/\`, \`data/chunks/\` are intentionally not committed
- ingestion tests rely on small synthetic RTF fixtures with no committed output
- PR verification notes say "ran locally" with no inspectable artifact

"Tests passed" is not sufficient evidence for pipeline PRs. A reviewer needs to see what the pipeline actually produces on real-shaped input — without access to the private corpus.

### 1. Committed fixture corpus (`tests/fixtures/srd_35/`)

Create a small set of synthetic RTF fixtures that are safe to commit and representative enough to exercise known edge cases:

- heading splitting (multi-section document)
- list-like content (numbered and bulleted)
- table-like content (cell-delimited rows)
- a slightly messy case (mixed formatting, near-heading false positives)
- a file with no detectable headings (whole-file fallback)

Fixtures must be minimal but non-trivial — richer than the inline fixture strings currently in `setUp()`.

### 2. Committed golden outputs

For the fixture corpus, commit expected pipeline outputs at each stage:

- `tests/fixtures/expected/extracted/` — plain text for each fixture
- `tests/fixtures/expected/canonical/` — canonical JSON for each fixture
- Later (chunking stage): `tests/fixtures/expected/chunks/`

Golden files exist so that a diff in a PR immediately shows behavioral change. They do not need to cover the private corpus.

### 3. Golden comparison in the test suite

Add a test class (e.g., `test_golden_ingestion.py`) that:

- runs the ingestion pipeline over the committed fixtures
- compares actual output against committed golden files
- fails with a readable diff on mismatch
- supports a `--update-golden` flag (or env var) to regenerate when behavior change is intentional

### 4. Human-readable preview artifact

Add a script (e.g., `scripts/preview_fixtures.py`) that renders a Markdown report for the fixture corpus showing:

- fixture filename and raw RTF snippet
- extracted plain text
- canonical document(s): `document_id`, `section_path`, `source_location`, `content` preview
- section count per file

Output should be committable as `tests/fixtures/PREVIEW.md` and regenerated as part of the evidence step. It should be easy to open during PR review with no tooling other than a browser or text editor.

### 5. PR evidence expectation (documented)

Add a short section to `docs/roadmap.md` (or a new `docs/pr_evidence.md`) defining the minimum evidence standard for pipeline PRs:

| PR type | Required evidence |
|---|---|
| Ingestion change | golden diff or preview regenerated and committed |
| Chunking change | golden chunk diff + preview section |
| Retrieval change | end-to-end example on fixture corpus |
| Schema change | updated golden outputs + validation gate result |

A pipeline PR that changes behavior without an updated preview or golden diff is incomplete.

## Done when

- `tests/fixtures/srd_35/` contains at least 5 committed synthetic RTF fixtures covering the cases above
- `tests/fixtures/expected/` contains committed extracted and canonical golden outputs for those fixtures
- a golden comparison test runs in the existing test suite and fails on behavioral regression
- `scripts/preview_fixtures.py` generates `tests/fixtures/PREVIEW.md` and can be regenerated locally with one command
- PR evidence expectations are documented in at least one committed doc file

## Non-goals

- does not require committing private corpus files
- does not replace existing unit tests
- does not require full benchmark or eval infrastructure
- preview artifact does not need to be a web app or CI-hosted page

## Blocks

- #5 (end-to-end baseline) — chunking and retrieval PRs cannot meet the evidence standard without this fixture layer in place

### PR DESCRIPTION
**Pull request by WanxiaJaneYang**
## Summary

Implements issue #17. Adds the committed evidence layer for pipeline PRs so reviewers can inspect actual behavior, not just code and unit test status.

### 1. Committed fixture corpus (`tests/fixtures/srd_35/`)

Five synthetic RTF fixtures, each covering a distinct pipeline edge case:

| File | Edge case exercised |
|---|---|
| `CombatActions.rtf` | Multi-section document — heading splitting |
| `SkillDescriptions.rtf` | List and tab-separated DC table |
| `MonsterStats.rtf` | RTF `\cell` → ` | ` separator (table content) |
| `SpellComponents.rtf` | Near-heading false positives (`Evocation [Fire]`, `V, S, M`) |
| `BasicMechanics.rtf` | No headings — whole-file fallback behavior |

### 2. Committed golden expected outputs (`tests/fixtures/expected/`)

- `extracted/text/*.txt` — plain text output per fixture
- `canonical/*.json` — canonical document per fixture (`ingested_at` normalised to `__TIMESTAMP__`)

Golden diff in a PR = direct evidence of what the behavior change produces.

### 3. Golden comparison test (`tests/test_golden_ingestion.py`)

10 tests. Runs the pipeline over the fixture corpus and compares against golden files:
- exact extracted text match
- canonical JSON structural match (excluding timestamp)
- required provenance fields (`document_id`, `locator`, `source_checksum`, etc.)
- per-fixture behavioral assertions (section headings present, tabs preserved, cell separators, etc.)

Failure is intentionally verbose — the message tells you exactly what changed and how to regenerate.

### 4. Preview script (`scripts/preview_fixtures.py`)

Generates `tests/fixtures/PREVIEW.md` — a committed Markdown file showing RTF snippet → extracted text → canonical metadata for each fixture. Reviewable directly in GitHub without running anything.

Regenerate golden files + preview in one command:
```bash
python scripts/preview_fixtures.py --update-golden
```

### 5. PR evidence standard (`docs/pr_evidence.md`, `docs/zh/pr_evidence.md`)

Documents the minimum evidence per PR type:

| PR type | Required evidence |
|---|---|
| Ingestion behavior change | Updated golden diff + regenerated PREVIEW.md committed in same PR |
| Chunking behavior change | Updated golden chunk files + preview section |
| Retrieval behavior change | End-to-end example on fixture corpus in PR description |
| Schema change | Updated golden canonical files + `--require-schema-validation` output |
| RTF decoder change | Golden extracted text diff for affected fixtures |

## Verification

```
python -m unittest tests.test_golden_ingestion tests.test_ingest_srd_35 tests.test_fetch_srd_35 -v
```

**25 tests, all pass.**

```
python scripts/preview_fixtures.py
```

Regenerated `tests/fixtures/PREVIEW.md` — open it to inspect fixture behavior.

## Note for PR #16

When PR #16 (sectioning + hardening) merges, its pipeline changes will cause golden test failures by design. The PR #16 author should run:

```bash
python scripts/preview_fixtures.py --update-golden
```

and commit the updated golden diff as evidence of the sectioning behavior change.

Closes #17

### CODE REVIEW

**Review by WanxiaJaneYang — COMMENTED** [review]
## Review summary

Good design intent and well-written evidence standard docs. Four issues to fix before merge, plus there are now conflicts with master (PRs #16 and #19 have landed).

### Issues

**Issue 1 — `_run_pipeline_to_temp` is dead code with a correctness bug** (`scripts/preview_fixtures.py`)

The function is never called. If it were called it would return paths inside a `with tempfile.TemporaryDirectory()` block — the temp dir is deleted the moment the `return` exits the `with` block, so the caller receives stale paths. The `tmp_manifest` variable inside it is also constructed but never used. Remove the function.

**Issue 2 — Golden test re-implements the pipeline instead of calling it** (`tests/test_golden_ingestion.py`)

`_run_pipeline()` manually reconstructs canonical documents using private internals (`_sha1_bytes`, `_sanitize_identifier`, etc.) instead of calling `ingest_source()`. If the production code path changes, the golden tests can still pass while real behavior diverges — the opposite of what golden tests are for. The test should drive the production pipeline and compare its output to committed golden files.

Note: master already has the correct pattern for this — `run_fixture_ingestion` / `load_golden_outputs` in the package API.

**Issue 3 — Pipeline construction is duplicated in three places**

The canonical document dict (same fields, same logic) is built independently in `update_golden()`, `_run_pipeline_direct()`, and `test_golden_ingestion._run_pipeline()`. A field change requires three coordinated edits and will drift. Extract one shared helper.

**Issue 4 — SpellComponents per-fixture assertion doesn't test the documented edge case**

The fixture is described as testing 'near-heading false positives' (`Evocation [Fire]`, `V, S, M`). The assertion only checks that both spell names are present — it doesn't verify that `Evocation [Fire]` and `V, S, M` were not treated as standalone headings. The golden match catches it implicitly, but the named assertion should say what it protects.

### Conflicts

`docs/roadmap.md`, `docs/zh/roadmap.md`, `scripts/preview_fixtures.py`, `tests/fixtures/PREVIEW.md`, and `tests/test_golden_ingestion.py` all conflict with master. Master's versions of the script and test files are already the corrected pattern (issues 2 and 3 above are resolved there). Rebasing and taking master's implementations for those files will collapse most of the code issues.

File: scripts/preview_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        abs_manifest["local_layout"]["canonical_root"] = str(tmp_path / "canonical")
+
+        # Patch resolve to allow paths outside repo root by running with tmp as a fake repo root.
+        extracted_abs = tmp_path / "extracted"
+        canonical_abs = tmp_path / "canonical"
+        extracted_abs.mkdir(parents=True, exist_ok=True)
+        canonical_abs.mkdir(parents=True, exist_ok=True)
+
+        _run_pipeline_direct(manifest, repo_root, extracted_abs, canonical_abs)
+        return extracted_abs, canonical_abs
```
Comment (WanxiaJaneYang) [inline scripts/preview_fixtures.py]: `_run_pipeline_to_temp` is never called from `main()` and has a correctness bug: the `return` executes *inside* the `with tempfile.TemporaryDirectory()` block, so the temp dir is deleted before the caller can use the returned paths. `tmp_manifest` is also built but immediately overridden by `abs_manifest`. Remove this function entirely.

File: tests/test_golden_ingestion.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _run_pipeline(self) -> dict:
+        from scripts.ingest_srd35.paths import resolve_repo_relative_path
+        from scripts.ingest_srd35.pipeline import build_source_ref, _sha1_bytes, _sanitize_identifier
+        from scripts.ingest_srd35.constants import EXTRACTION_CAVEATS, INGESTION_NOTES
+        from datetime import datetime, timezone
+
```
Comment (WanxiaJaneYang) [inline tests/test_golden_ingestion.py]: `_run_pipeline()` reimplements the canonical document construction by hand (importing private `_sha1_bytes`, `_sanitize_identifier`, etc.) instead of calling `ingest_source()`. A pipeline change can make this re-implementation diverge silently — the golden tests pass while real behavior changes, which defeats the purpose of golden tests. The test should drive the production pipeline and compare its output to the committed golden files. Master already has the correct pattern: `run_fixture_ingestion` / `load_golden_outputs` in the package API.

File: tests/test_golden_ingestion.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            f"Remove stale golden files or restore the fixture.",
+        )
+
+    # --- Per-fixture behavioral assertions ---
+
+    def test_combat_actions_multi_section_content_preserved(self) -> None:
+        """CombatActions fixture must contain all three section headings in extracted text."""
+        decoded = self._run_result["combatactions"]["decoded"]
+        for heading in ("Full Attack", "Standard Action", "Move Action"):
+            self.assertIn(heading, decoded, f"Missing section heading: {heading}")
```
Comment (WanxiaJaneYang) [inline tests/test_golden_ingestion.py]: The fixture is documented as testing "near-heading false positives" (`Evocation [Fire]`, `V, S, M`). The assertion only checks that both spell names are present — it does not verify that `Evocation [Fire]` and `V, S, M` were *not* treated as standalone headings/section splits. Add an assertion like `self.assertNotIn("\n\nEvocation [Fire]\n\n", decoded)` to pin the false-positive guard.

**Comment by WanxiaJaneYang** [comment]
## Changes made to address review + resolve conflicts

**Rebased onto master** (which had PRs #16 and #19 merged in, each of which added real SRD fixtures and production-grade evidence tooling).

### Conflicts resolved

| File | Resolution |
|---|---|
| `scripts/preview_fixtures.py` | Took master's version — uses `run_fixture_ingestion`/`write_golden_outputs` from the pipeline package; no dead code |
| `tests/test_golden_ingestion.py` | Took master's version — calls production pipeline directly via `run_fixture_ingestion` + `load_golden_outputs` (issues 2 and 3 from review are now resolved) |
| `tests/fixtures/PREVIEW.md` | Took master's version — generated from real SRD fixtures |
| `docs/roadmap.md` | Merged both sides: kept master's checked items, added path annotations from this branch |
| `docs/zh/roadmap.md` | Same merge strategy |

### Stale files removed

- **Synthetic fixture RTFs** (`BasicMechanics.rtf`, `CombatActions.rtf`, `MonsterStats.rtf`, `SkillDescriptions.rtf`, `SpellComponents.rtf`) — master already has real SRD fixture files that cover the same edge cases
- **Synthetic golden expected files** (5 canonical JSONs + 5 extracted text files) — wrong format for master's section-level pipeline output
- **`tests/fixtures/manifest.json`** — not used by master's `run_fixture_ingestion` (which builds its manifest internally)
- **`docs/pr_evidence.md`** and **`docs/zh/pr_evidence.md`** — master already has the canonical version at `docs/standards/pr_evidence.md` (from PR #19)

### What this PR now contributes

Path annotations on two already-checked roadmap items, pointing to the canonical locations of what was implemented. All 28 tests pass.
